x-pack/plugin/inference/src/internalClusterTest/java/org/elasticsearch/xpack/inference/integration/MockInferenceServiceIT.java

```diff
@@ -79,6 +83,24 @@ public void testMockService() {
         inferOnMockService(modelId, TaskType.SPARSE_EMBEDDING, randomAlphaOfLength(10));
     }
 
+    public void testMockService_DoesNotReturnSecretsInGetResponse() throws IOException {
```

Comment:
I wasn't quite sure how to test that the secrets don't get returned since they're not included in the class definition. Let me know if this is what you were thinking.

Reply:
This test is an integration test (extends `ESIntegTestCase`), inside this test the code has access to the internal client and can call Transport Actions directly.

The main other type of test is a rest test (extends `ESRestTestCase`), rest tests run against an external cluster at the rest boundary and don't have access to the internal client. The yaml tests are rest tests.

This test is fine and the coverage is good but we should also add a rest test to prove that when called the `GET _inference/task_type/model` API (from outside the cluster not using the internal client) the response does not contain any secrets. We know it doesn't because the class used in the response does not contain any secrets but good to have a test showing that anyway.

Reply:
`PyTorchModelIT` is an example of a rest test, to create one in this project the following must be added to `build.gradle`

```apply plugin: 'elasticsearch.java-rest-test'```

If you add a rest test in the same package as `MockInferenceServiceIT` invoke it with `./gradlew :x-pack:plugin:inference:internalClusterTest`